search-provider/eks-discovery-feed-provider.c

```diff
@@ -435,6 +469,166 @@ handle_article_card_descriptions (EksDiscoveryFeedDatabaseContentProvider *skele
 }
 
 static void
+get_word_of_the_day_content_cb (GObject *source,
+                                GAsyncResult *result,
+                                gpointer user_data)
+{
+  EkncEngine *engine = EKNC_ENGINE (source);
+  DiscoveryFeedQueryState *state = user_data;
+
+  g_application_release (g_application_get_default ());
+
+  GError *error = NULL;
+  g_autoptr(GSList) models = NULL;
+  GSList *shards = NULL;
+
+  if (!models_and_shards_for_result (engine,
+                                     state->provider->application_id,
+                                     result,
+                                     &models,
+                                     &shards,
+                                     &error))
+    {
+      g_dbus_method_invocation_take_error (state->invocation, error);
+      discovery_feed_query_state_free (state);
+      return;
+    }
+
+  GVariantBuilder builder;
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{ss}"));
+
+  gint index = get_day_of_year () % g_slist_length (models);
+  EkncContentObjectModel *model = g_slist_nth (models, index)->data;
+
+  add_key_value_pair_from_model_to_variant (model, &builder, "title");
+  add_key_value_pair_from_model_to_variant (model, &builder, "synopsis");
+  add_key_value_pair_from_model_to_variant (model, &builder, "ekn-id");
```

Comment:
Right the shard does not have those fields atm:

`{"contentType": "text/html", "license": "noun", "sequenceNumber": 0, "title": "nosegay", "outgoingLinks": [], "lastModifiedDate": "2017-04-30T04:00:00.000Z", "tags": ["April-86", "word_of_day", "EknArticleObject"], "matchingLinks": ["http://www.thefreedictionary.com/_/archive.htm?d=4/30/2017"], "originalTitle": "nosegay", "synopsis": "A small bunch of flowers.", "originalURI": "http://www.thefreedictionary.com/_/archive.htm?d=4/30/2017", "isServerTemplated": true, "featured": false, "@id": "ekn://ae41c6b0-2ea4-11e7-9859-5d33d75bbfc1/3c8ffb0e67614f8c10ca82380e672726c40cda08", "@type": "ekn://_vocab/ArticleObject", "resources": []}`

We need to add those first.